dali/pipeline/data/buffer.h

```diff
@@ -377,6 +438,7 @@ class DLL_PUBLIC Buffer {
     }
   }
 
+ protected:
```

Comment:
Since you're removing inheritance from `Buffer`, I'd make the `move_buffer` method private. Whoever would like to inherit from `Buffer` in the future would need to write his own move routine anyway, so it's good idea to restrict this only for `Buffer` usage.

Or maybe remove the `move_buffer` and implement it in the idiomatic way?
```c++
operator=(Buffer&&) {
<what's currently in move_buffer(...)>
}

Buffer(Buffer&& b) {
*this = move(b);
}
```

Reply:
I will probably do a similar PR just for Tensor, I can adjust it there.

Reply:
I'd keep it protected. This method is safe and self-contained. There's no need to make it public, since it has little merit outside of inheritance context (it's preferable to just move the entire thing), but when implementing move semantics in derived classes, it's much more readable to write `move_buffer(other)` than to resort to monstrosities like:
`static_cast<Buffer<Backend>&>(*this) = std::move(static_cast<Buffer<Backend>&>(other));`